lib/styles.js

```diff
@@ -340,4 +342,133 @@ module.exports = function (expect) {
             this.sp().commentBlock(pen);
         }
     });
+
+    expect.addStyle('diffFragment', function (text, baseStyle, specialCharStyle, markUpSpecialCharacters) {
+        if (markUpSpecialCharacters) {
+            text.split(specialCharRegExp).forEach(function (part) {
+                if (specialCharRegExp.test(part)) {
+                    this[specialCharStyle || baseStyle](utils.escapeChar(part));
+                } else {
+                    this[baseStyle](part);
+                }
+            }, this);
+        } else {
+            this[baseStyle](text);
+        }
+    });
+
+    expect.addStyle('stringDiff', function (actual, expected, options) {
+        options = options || {};
+        var type = options.type || 'WordsWithSpace';
+        var diffLines = [];
+        var lastPart;
+        stringDiff.diffLines(actual, expected).forEach(function (part) {
```

Comment:
This is the part I don't really like, the style does the actual diff computation instead of being handed the diff. I know that the API from the outside would be convenient by letting the style taking care of it, but I just think it is misplaced. I would prefer:

```js
var diff = utils.stringDiff(actual, expected)
pen.appendDiff(diff)
```

It is mainly about separation of concerns.